loopd/swapclient_server.go

```diff
@@ -990,9 +999,52 @@ func validateLoopOutRequest(chainParams *chaincfg.Params, confTarget int32,
 	}
 
 	// Check that the label is valid.
-	if err := labels.Validate(label); err != nil {
+	if err := labels.Validate(req.Label); err != nil {
+		return 0, err
+	}
+
+	channels, err := lnd.ListChannels(ctx)
+	if err != nil {
 		return 0, err
 	}
 
-	return validateConfTarget(confTarget, loop.DefaultSweepConfTarget)
+	outgoingChanSetMap := make(map[uint64]bool)
+	for _, c := range req.OutgoingChanSet {
+		outgoingChanSetMap[c] = true
+	}
+
+	var totalOutgoingCapacity btcutil.Amount
+	for _, c := range channels {
+		// Don't bother adding the local balance of an inactive channel.
+		if !c.Active {
+			continue
+		}
+
+		// If no outgoing channel set was given then the entire set of
+		// active channels is considered.
+		if len(outgoingChanSetMap) == 0 {
+			totalOutgoingCapacity += c.LocalBalance
+			continue
+		}
+
+		// If a channel set was specified then only the specified
+		// channels are considered.
+		if !outgoingChanSetMap[c.ChannelID] {
+			continue
+		}
+
+		totalOutgoingCapacity += c.LocalBalance
+	}
+
+	// Determine if the total outgoing capacity of the channel set is
+	// enough to satisfy the loop amount and the maximum possible fees
+	// that could be charged.
+	if req.Amt+req.MaxSwapRoutingFee+req.MaxSwapFee >
```

Comment:
Not 100% sure if these are all the fees that need to be considered... only concerned with what the channel set needs to supply. Should the reserve amount (the amount that needs to be kept on your side of the channel also be considered? I feel like it should but not sure where to get this value/percentage. 

Once the correct fees and channel reserve properties are taken into account, i will add tests that test for those too.

Reply:
Reserve amount is listed separately in listchannels, so you don't need to worry about that here. All the local balance is spendable.

We only need to be able to route the request amount + maxrouting, because the swap fee is included in the amount routed, and then we just pay out less on chain.